### PR TITLE
Ft/m UI update

### LINKED ISSUE #67: Upgrade MUI to v5
Upgrade to the latest MUI version (v5)

### PR DESCRIPTION
**Pull request by kelvinkipruto**
## Description

Update MUI to v5

Fixes #67 

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -7,7 +7,6 @@
     "simplebar-react": "^2.1.0"
   },
   "peerDependencies": {
-    "@material-ui/core": "^4.9.3",
```
Comment (kilemensi) [inline packages/core/package.json:10]: This should be replace with `@mui/material`, no?

File: packages/core/src/DataSourceContent/useStyles.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
-import { makeStyles } from "@material-ui/core/styles";
+import { makeStyles } from 'tss-react/mui';
```
Comment (kilemensi) [inline packages/core/src/DataSourceContent/useStyles.js]: `tss-react`?

**Review by kilemensi — COMMENTED** [review]
👍🏽 @kelvinkipruto 

----

Just had a quick look but I'm not really sure we need to surround every component with
```jsx
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={theme}>
```

Can you may be point to docs or something that can give the reviewers necessary background context?

**Comment by kelvinkipruto** [comment]
@kilemensi ` <StyledEngineProvider injectFirst>` was automatically added when running the migration [codemod](https://mui.com/material-ui/migration/migration-v4/#preset-safe)

File: packages/core/package.json (hunk 1 above)
Comment (kelvinkipruto) [reply packages/core/package.json:10]: As per the [docs](https://mui.com/material-ui/migration/migration-v4/#:~:text=%40material%2Dui/core%20%2D%3E%20%40mui/material), I think so.

File: packages/core/src/DataSourceContent/useStyles.js (hunk 2 above)
Comment (kelvinkipruto) [reply packages/core/src/DataSourceContent/useStyles.js]: This was the best option I found that would not require a lot of changes. It is also part of the [docs](https://mui.com/material-ui/migration/migrating-from-jss/#2-use-tss-react)

File: stories/Section.stories.js [hunk 3]
```diff
@@ -38,7 +38,9 @@ const Template = ({ fixed, ...args }) => {
     },
   };
 
-  const classes = makeStyles(({ breakpoints }) => ({
+  // TODO jss-to-tss-react codemod: Unable to handle style definition reliably. Unsupported arrow function syntax.
+  // Unexpected value type of CallExpression.
+  const classes = makeStyles()(({ breakpoints }) => ({
```
Comment (kilemensi) [inline stories/Section.stories.js]: What's happening here?

File: package.json [hunk 4]
```diff
@@ -49,7 +52,7 @@
     "react-is": "^17.0.2",
     "regenerator-runtime": "^0.13.5",
     "simplebar-react": "^2.1.0",
-    "storybook-addon-material-ui": "^0.9.0-alpha.24",
```
Comment (kilemensi) [inline package.json:52]: Why are we dropping this?

File: packages/core/package.json (hunk 1 above)
Comment (kilemensi) [inline packages/core/package.json:10]: If we don't included `@mui/material` here, how does core import `@mui/material` components?

**Review by kilemensi — COMMENTED** [review]
👍🏽 ... Looks good, still need to check the tss-react in details though.

File: package.json (hunk 4 above)
Comment (kelvinkipruto) [reply package.json:52]: It was needed to create the theme for storybook. We've now replaced it [MUI ThemeProider](https://github.com/CodeForAfrica/COMMONS-UI/pull/68/files#:~:text=()%3B-,export%20const%20decorators%20%3D%20%5BmuiTheme(%5Btheme%5D),%5D%3B,-const%20size%20%3D)

File: packages/core/package.json (hunk 1 above)
Comment (kelvinkipruto) [reply packages/core/package.json:10]: I agree. Let me add this.

**Comment by kelvinkipruto** [comment]
> 👍🏽 ... Looks good, still need to check the tss-react in details though.

[Here](https://mui.com/material-ui/migration/migrating-from-jss/#2-use-tss-react) is the official MUI docs about it

**Comment by kilemensi** [comment]
QQ @kelvinkipruto. Why are we changing
```jsx
 const classes = useStyles(props);
```
to
```jsx
 const classes = useStyles(props, {
    props,
  });
```
?

File: packages/core/src/Layout/index.js [hunk 5]
```diff
@@ -15,7 +15,11 @@ const Layout = React.forwardRef(function Layout(
   { children, classes: classesProp, className, ...props },
   ref
 ) {
-  const classes = useStyles({ classes: classesProp });
+  const classes = useStyles(undefined, {
+    props: {
+      classes: classesProp,
+    },
+  });
```
Comment (kilemensi) [inline packages/core/src/Layout/index.js]:
```suggestion
    const classes = useStyles({ classes: classesProp });
```

**Review by kilemensi — COMMENTED** [review]
👍🏽 

----

I think some items are broken (ScrollBar components, DataSourceContent). Compare them with https://codeforafrica.github.io/COMMONS-UI/.

Conform if any of the broken components are used downstream in PesaYetu.

**Comment by kelvinkipruto** [comment]
@kilemensi None of them are used in PesaYetu

**Comment by kelvinkipruto** [comment]
@kilemensi Should I proceed to merge this?

**Comment by kilemensi** [comment]
If all errors are coming from upstream PesaYetu @kelvinkipruto , then yes.

**Pull request merged**